hazelcast/src/main/java/com/hazelcast/map/impl/tx/TransactionalMapProxySupport.java

```diff
@@ -280,25 +281,26 @@ private VersionedValue lockAndGet(Data key, long timeout) {
     }
 
     protected Set queryInternal(final Predicate predicate, final IterationType iterationType, final boolean dataResult) {
-        final NodeEngine nodeEngine = getNodeEngine();
+        NodeEngine nodeEngine = getNodeEngine();
         OperationService operationService = nodeEngine.getOperationService();
         Collection<MemberImpl> members = nodeEngine.getClusterService().getMemberList();
         int partitionCount = nodeEngine.getPartitionService().getPartitionCount();
         Set<Integer> partitions = new HashSet<Integer>(partitionCount);
         QueryResultSet result = new QueryResultSet(nodeEngine.getSerializationService(), iterationType, dataResult);
-        List<Integer> missingList = new ArrayList<Integer>();
+
         try {
             List<Future> futures = new ArrayList<Future>();
-            invokeQueryOperaration(predicate, operationService, members, futures);
+            invokeQueryOperation(predicate, operationService, members, futures);
             collectResults(partitions, result, futures);
             if (partitions.size() == partitionCount) {
                 return result;
             }
-            findMissingPartitions(partitionCount, partitions, missingList);
         } catch (Throwable t) {
-            missingList.clear();
-            findMissingPartitions(partitionCount, partitions, missingList);
+            EmptyStatement.ignore(t);
```

Comment:
`missingList` is just used as parameter of `findMissingPartitions()`. So before `findMissingPartitions()` is called the list is always empty.

Since `findMissingPartitions()` is the last statement in the `try` block there can be nothing in `missingList` if an exception occurs from the other statements. So the `misslingList.clear();` seems pointless.

If `findMissingPartitions()` is causing the exception it will happen again, since we call it with the same parameters, which cannot change in between. And since there is no other `try`-`catch` the exception of the second call would be propagated.

So it should be the same behavior (just easier to read) to create an fill the list after the `try` block.